### PR TITLE
Submission external sources limited on metadata field

### LINKED ISSUE #943: External Sources cannot be limited based on metadata field
**Describe the bug**
Discovered in review of https://github.com/DSpace/dspace-angular/pull/932#pullrequestreview-525533457

When a Lookup button is added to a metadata field (e.g. Author field), there is currently no way to limit which External Sources are searched/searchable.  

This can result in very odd/unexpected behavior like this:
1. Ensure the Author field has a Lookup button, which is the case at https://dspace7-demo.atmire.com/home
2. Click the Lookup and search PubMed for "math"
3. Select an article and click Import

Strangely, this all works...the system lets you import an Article via the Author field.  The Article never appears in the author field, but if you click Lookup again, you'll see the "Current Selection" tab lists the Article as an Author.  **This has the greatest impact on Configurable Entities as it lets you create incorrect *relationships*...in this case you are creating an Author based relationship to an Article/Publication.**

**Expected behavior**
I'd expect that there should be a way to configure which External Sources are used based on the field "type".  For example, in the case of an Author field, it only makes sense to search "Local Authors", "ORCID" and "LC Names".  

It's possible (but not necessarily required) this could be made configurable as part of the Submission configuration eventually.  But, initially we need to at least ensure you cannot

### PR DESCRIPTION
**Pull request by Atmire-Kristof**
## References
- Fixes #943
- Depends on #975
- The configured REST API requires [PR #3109](https://github.com/DSpace/DSpace/pull/3109) for the external sources to be limited.

## Description
Currently, the lookup window for relationships during submission displays all the available external sources to import from, even sources that don't match the field you're trying to import a relationship to.
This PR aims to only display only the configured external sources depending on the [new configuration returned by the REST API](https://github.com/DSpace/DSpace/pull/3109), as well as display error notifications when creating the relationship fails.

This PR seems a lot bigger than it is, because it depends on #975 You can see only the changes in this PR [here](https://github.com/atmire/dspace-angular/compare/cache-redesign-part-2...atmire:Submission-external-sources-limited-on-metadata-field)

## Instructions for Reviewers
Changes made:
- Added `externalsources` array property to `RelationshipOptions` (retrieved from REST API)
- Modified lookup modal to only retrieve the appropriate external source objects depending on this new property
- RelationShipService's `getRelationshipTypeByLabelAndTypes()` now emits "null" whenever the types found in the created item and/or submission item don't match the relationship.
- A notification is displayed upon creating a relationship where this method emits "null". The newly created item is also automatically deselected if this is the case.

How to test:
- Configure a REST API containing [PR #3109](https://github.com/DSpace/DSpace/pull/3109), as well as a collection with a submission form containing a relationship metadata field with a custom configuration for the external sources.
- Visit the submission form of this collection
- Open the lookup modal for the relationship metadata field containing configured external sources
- Verify only the configured external sources are available
- Verify you're able to import an item from the external source and add it as a relationship to the submission item

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [TSLint](https://palantir.github.io/tslint/) validation using `yarn run lint`
- [x] My PR doesn't introduce circular dependencies
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.

### CODE REVIEW

**Comment by tdonohue** [comment]
@Atmire-Kristof : Please rebase on latest `main`.  Keep in mind we just updated to Angular 10, which changed some lint rules. So, you may also want to ensure `yarn lint` doesn't throw errors (as its possible the PR may need minor formatting updates to pass).

**Comment by tdonohue** [comment]
@Atmire-Kristof : Now that #975 has been merged, please rebase this PR so that it only shows the changes that are new/unique.  Thanks!

**Review by tdonohue — CHANGES_REQUESTED** [review]
@Atmire-Kristof : Overall, this PR works as described.  However, there are some minor changes needed to the code before we can move this along.

Additionally, I noticed that the External Sources popup modal in the Submission is showing empty lists of results -- so, for example, you can use the Author lookup and it only searches ORCID, but the results that come back have no information/titles (only import buttons are shown).  However, this bug appears to exist also on the demo site (https://demo7.dspace.org/) so it doesn't appear to be a new bug caused by this PR. I only mention it in case it's easy to resolve as part of this PR...if not, I can log a separate ticket.

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "dspace-angular",
-  "version": "0.0.0",
+  "version": "1.22.10",
```
[the remark this answers is not in the document]
Comment (Atmire-Kristof) [reply package.json]: @tdonohue , I don't remember why I made this change, but I do vaguely remember my messing around with my node version that day, so perhaps it's something that happened automatically without me noticing. Reverted the version back to 0.0.0.

**Comment by Atmire-Kristof** [comment]
> @Atmire-Kristof : Overall, this PR works as described. However, there are some minor changes needed to the code before we can move this along.
> 
> Additionally, I noticed that the External Sources popup modal in the Submission is showing empty lists of results -- so, for example, you can use the Author lookup and it only searches ORCID, but the results that come back have no information/titles (only import buttons are shown). However, this bug appears to exist also on the demo site (https://demo7.dspace.org/) so it doesn't appear to be a new bug caused by this PR. I only mention it in case it's easy to resolve as part of this PR...if not, I can log a separate ticket.

@tdonohue , thanks for the feedback. The issue with the items in the external sources popup modal should now be fixed. The submission module was missing imports for the JournalEntitiesModule and ResearchEntitiesModule, causing every item in the list to be treated as a regular Publication instead of their actual relationship type.

**Review by tdonohue — APPROVED** [review]
👍 I retested today, and it's all working great now @Atmire-Kristof !  Thanks!  All my prior feedback was also addressed.

**Pull request merged**